pkg/core/xds/merge.go

```diff
@@ -42,18 +44,123 @@ func MergeConfs(confs []interface{}) (interface{}, error) {
 		return nil, err
 	}
 
+	valueResult := reflect.ValueOf(result)
 	// clear appendable slices, so we won't duplicate values of the last conf
-	clearAppendSlices(reflect.ValueOf(result))
+	clearAppendSlices(valueResult)
 	for _, conf := range confs {
 		// call .Elem() to unwrap interface{}
-		appendSlices(reflect.ValueOf(result), reflect.ValueOf(&conf).Elem())
+		appendSlices(valueResult, reflect.ValueOf(&conf).Elem())
 	}
 
-	v := reflect.ValueOf(result).Elem().Interface()
+	if err := handleMergeByKeyFields(valueResult); err != nil {
+		return nil, err
+	}
+
+	v := valueResult.Elem().Interface()
 
 	return v, nil
 }
 
+type acc struct {
+	Key      interface{}
+	Defaults []interface{}
+}
+
+const (
+	defaultFieldName = "Default"
```

Comment:
Maybe we should use a `mergeValue` tag instead. It will be more clear when you're reading the policy definition:
```go
type mergeEntry struct {
	Key     mergeKey  `json:"key" xdsMerge:"mergeKey"`
	Default mergeConf `json:"default" xdsMerge:"mergeValue"`
}
```
So you don't have to know the assumption that `Default` is a special name for merging

Reply:
Plus there's room for `Override` if we add support for that.